doc/src/how_to_guides/index.rst

```diff
@@ -0,0 +1,15 @@
+.. _how_to_guides:
+
+====================
+ SymPy How-To Guides
+====================
+
+The purpose of this collection of documents is to provide users of SymPy with
+topics which are not strictly tutorial or are longer than tutorials and tests.
+The documents will hopefully show how SymPy can be used in more advanced topics.
```

Comment:
I don't understand this explanation or reasoning for the new "how-to guides" section. We have a single sympy tutorial: https://docs.sympy.org/latest/tutorial/index.html (I think that is the only one), so I don't know what "tutorials" refers to.

There are other tutorial-like things, like `numeric-computation.rst`, or maybe you want to call them "how to guides". https://docs.sympy.org/latest/gotchas.html also seems like a "how-to guide".

We also have documentation generated from the `__doc__` attributes of modules, functions, classes, methods, etc. Some of the modules `__doc__` text is also very much like a tutorial/how-to guide (e.g. https://docs.sympy.org/latest/modules/printing.html).

We have things like this too: https://docs.sympy.org/latest/modules/physics/vector/vectors.html which are reference documents or maybe a how-to guide or something else.

I agree that we need some new organization wrt these types of docs, but I'm not sure adding a "How-To Guides" section is the best solution, especially when some of the other narrative docs are not included there.

The `__doc__` sections for classes, methods, functions, attributes now have to follow the SymPy docstring style guide. These can then clearly be separated out into "API documentation" (much like most other libraries do). The `__doc__` of many modules is a bit more confusing and some may belong in `API documentation` but some may not.

I've used the word "narrative" for describing the non-api docs in the past. Anything where you are writing sentences, paragraphs, mixed with code for explanation. Can we have only three headings in the top-level table of contents like:

```
Narrative
=======

   install.rst
   tutorial/index.rst
   gotchas.rst
   special_topics/index.rst
   documentation-style-guide.rst

API
===
   
modules/index.rst

Other
=====
   wiki.rst
   outreach.rst
   aboutus.rst
   citing.rst

```

The "other" section has things I think belong on the SymPy website, not necessarily in the docs. If so, we could remove that section and only have narrative and api sections.

Also, "narrative" is not a good name for the section header. Maybe "how-to guides" is better or something else. Or just "Guides".

The tricky thing is that there are still narrative documentation mixed into the modules/ folder. The https://docs.sympy.org/latest/modules/physics/vector/vectors.html is an example. It is unclear we we should extract those so they are separated from the API docs. I think most nice documentation websites do that.

I've written about this before in other email, github, etc. posts. I personally, would like to see the API and narrative documentation separated.

Reply:
Pandas has 4 very clear divisions:

https://pandas.pydata.org/docs/

- Getting started
- User guide
- API
- Developer guide

That sounds nice to me. Maybe we could reorganize in that way?

Reply:
Perhaps, but it is too major to be done here. I reverted this change, but leaving this conversation open for future reference.

Reply:
@moorepants the distinction comes from https://documentation.divio.com/. I recommend watching the presentation on that page if you have the time. I believe NumPy also planning on using this division for their documentation. 

I agree "tutorials" plural in this text is a bit confusing. This is referring to the idea of tutorials in general, but we only have one tutorial for SymPy.

We absolutely need some sort of top-level section for guides. Whether it is called "how to guides" or something else I don't care. The problem is that such guides clearly don't belong in the API documentation. So when people want to add one, they look at what else is there and they see the tutorial. But the tutorial is NOT where these things should go either. The tutorial is meant to be an introductory tutorial for people new to SymPy, and these how-to guides are out of scope for it (again, please see the Divio presentation). But I've seen this happen a dozen times, including an earlier commit of this PR, where people want to add something that is essentially a how-to guide, and they try to put it in the tutorial, and I have to explain that it doesn't belong there. So we need some sort of obvious top level place in the docs where people will put these things that isn't the tutorial. I also hope having a clear place for these will encourage writing more long form non-API documentation, which is an area SymPy's documentation is weak in.

Reply:
Unfortunately @JSS95 force pushed so I can't read what was written here except for the blurb that GitHub shows for this comment. But based on what I can see here it looked fine to me, with maybe a few minor edits.

Reply:
I checked out the divio thing. It seems reasonable, although I'm not sure that description covers everything you might want in documentation. Regardless, I'm fine adapting that for SymPy. It would be a significant improvement to the organization of our documentation.